doc/configuration.md

```diff
@@ -28,6 +28,7 @@ The following options control the behavior of BuildCache:
 | `BUILDCACHE_PERF` | `perf` | Enable performance logging | false |
 | `BUILDCACHE_DISABLE` | `disable` | Disable caching (bypass BuildCache) | false |
 | `BUILDCACHE_ACCURACY` | `accuracy` | Caching accuracy (see below) | DEFAULT |
+| `BUILDCACHE_READONLY` | `read_only` | Only read and use the cache without updating it | false |
```

Comment:
This description looks very misleading. It does NOT mention that the command won't run on a miss. For me such a behavior is very unexpected (without reading the commit message, that is). I think this fact should be clearly explained (in the documentation). Even better it would be nice to use a more self-descriptive name for this mode.
`BUILDCACHE_FAIL_ON_MISS` or something.

Reply:
The description says that the cache is not updated in this mode, which means that the command will not be run on a miss. Or am I missing something?
I agree about a more self-descriptive name. But I haven't come up with a better idea yet. FAIL_ON_MISS - it would certainly have misled me.
If someone offers a better description or name I would be glad.

Reply:
Hm. It seems to me that the read-only mode *should* run the command on a miss, which is also the case in this implementation, or am I missing something?

Reply:
What does it mean to run the command on a miss? What is the command? That isn't obvious for me.

Reply:
"The command" is the actual command that BuildCache is wrapping (e.g. a compiler invocation). The normal flows are:

1. check cache -> cache hit -> use cached result
2. check cache -> cache miss -> run command -> cache the result of the command

With `BUILDCACHE_READONLY` set, we would instead have the following flows:

1. check cache -> cache hit -> use cached result
2. check cache -> cache miss -> run command (don't cache the result!)

(if I'm not mistaking)

Reply:
It's now clear.
You need a mode 'fail on a miss' (Stop build). Is it so?

Reply:
This configuration is fine, as is the implementation.

Reply:
One comment though: The env var is called `BUILDCACHE_READONLY` ("readonly" as a single word), while the JSON parameter is called `read_only` ("read only" as two words) and the config property is also called `read_only()`.

Please decide on one of the two. I prefer `BUILDCACHE_READ_ONLY`, since that is more consistent with the naming of other configuration parameters.

Reply:
done.